stdlib/gc.mli

```diff
@@ -145,17 +145,45 @@ type control =
     mutable allocation_policy : int;
     [@ocaml.deprecated_mutable
          "Use {(Gc.get()) with Gc.allocation_policy = ...}"]
-    (** The policy used for allocating in the heap.  Possible
-        values are 0 to 2.  0 is the original next-fit policy,
-        usually fast, but can result in fragmentation.  1 is the
-        first-fit policy, which can be slower in some cases but
-        can be better for programs with fragmentation problems.
-        2 is the best-fit allocation policy, the best for avoiding
-        fragmentation.
-        Note that changing the allocation policy at run-time forces
+    (** The policy used for allocating in the major heap.
+        Possible values are 0, 1 and 2.
+
+        - 0 is the next-fit policy, which is usually fast but can
+          result in fragmentation, increasing memory consumption.
+
+        - 1 is the first-fit policy, which avoids fragmentation but
+          has corner cases (in certain realistic workloads) where it
+          is sensibly slower.
+
+        - 2 is the best-fit policy, which is fast and avoids
+          fragmentation. In our experiments it is faster and uses less
+          memory than both next-fit and first-fit.
+
+        The current default is next-fit, as the best-fit policy is new
+        and not yet widely tested. We expect best-fit to become the
+        default in the future.
+
+        On one example that was known to be bad for next-fit and first-fit,
+        next-fit takes 28s using 855Mio of memory,
+        first-fit takes 47s using 566Mio of memory,
+        best-fit takes 27s using 545Mio of memory.
+
+        Note: the difference in fragmentation behavior means that
+        different policies will have different proportion of "wasted
+        space" for a given program. Less fragmentation means a smaller
+        heap so, for the same amount of wasted space, a higher
+        proportion of wasted space, making the GC work harder. When
+        changing to a low-fragmentation policy, you may need to
+        augment the [space_overhead] setting, for example using [100]
+        instead of the default [80] which tuned for next-fit.
```

Comment:
There is still `[80] which tuned` that looks like it should be `which is tuned`. I also wonder if the paragraph structure should be reversed: start with remarking that the `[space_overhead]` setting may need to be tuned differently for each policies and then explain why this fine tuning is dependent on the policy.

Reply:
Nice suggestion, thanks! Applied.